src/nock/recorder.ts

```diff
@@ -187,48 +197,65 @@ export async function endRecording({
 
   if (
     security.length > 0 ||
-    (integrationParameters && Object.values(integrationParameters).length > 0)
+    (integrationParameters &&
+      Object.values(integrationParameters).length > 0) ||
+    (inputVariables && Object.values(inputVariables).length > 0)
   ) {
-    checkSensitiveInformation(definitions, security, integrationParameters);
+    checkSensitiveInformation(
+      definitions,
+      security,
+      integrationParameters,
+      inputVariables
+    );
   }
 
-  const recordingExists = await exists(composeRecordingPath(recordingPath));
-
   if (recordingExists) {
     return await matchTraffic(recordingPath, definitions);
   }
 
   // recording file does not exist -> record new traffic
-  await writeRecordings(composeRecordingPath(recordingPath), definitions);
+  const path = composeRecordingPath(recordingPath);
+
+  await writeRecordings(path, definitions);
+
   debug('Recorded definitions written');
 
   return undefined;
 }
 
 export function composeRecordingPath(
   recordingPath: string,
-  version?: string
+  options?: {
+    version?: string;
+    decoded?: boolean;
+  }
 ): string {
-  if (version === 'new') {
-    return `${recordingPath}-new.json`;
+  if (options?.version === 'new') {
+    return `${recordingPath}-new${options.decoded ? '.decoded.json' : '.json'}`;
```

Comment:
One thing I don't understand: why we use `-` for `new` and `_` for other versions?

Reply:
BTW: this is breaking change, right? 🤔

Reply:
yes, but this versioning was not even in previous stable version